packages/generated-react/dist/eliza-BigIntService_connectquery_hooks.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// @generated by protoc-gen-connect-query-with-hooks v0.0.1 with parameter "target=ts,import-hook-from=@tanstack/react-query"
+// @generated from file eliza.proto (package buf.connect.demo.eliza.v1, syntax proto3)
+/* eslint-disable */
+// @ts-nocheck
+
+import { count } from "./eliza-BigIntService_connectquery.ts";
+import { UseBaseQueryOptions, useInfiniteQuery, UseInfiniteQueryOptions, useMutation, UseMutationOptions, useQuery } from "@tanstack/react-query";
+import { PartialMessage } from "@bufbuild/protobuf";
+import { CountRequest, CountResponse } from "./eliza_pb.js";
+import { ConnectError } from "@bufbuild/connect";
+
+/**
+ * @generated from rpc buf.connect.demo.eliza.v1.BigIntService.Count
+ */
+export const useCountQuery = 
+  ({
+    inputs,
+    transformParams,
+  }: {
+    inputs: Parameters<typeof count.useQuery>;
```

Comment:
I was probably a little hasty in building this API, given that this is what it would look like in usage:

```tsx
const q = useCountQuery({
  inputs: [{ count: 1 }],
});
```
for the common case. I think maybe we can do better for both common cases, and when you want to modify the params.

`useQuery` from `connect-query` always has two arguments, one for the input and the second is options. So maybe we do:

```tsx
export const useCountQuery = 
  (input: Parameters<typeof count.useQuery>[0], options: Parameters<typeof count.useQuery>[1], queryOptions?: {
    transformParams?: (
      baseOptions: ReturnType<typeof count.useQuery>
    ) => Partial<UseBaseQueryOptions<PartialMessage<CountRequest>, ConnectError>>;
  }) => {
    const { transformParams } = queryOptions ?? {};
    const baseOptions = count.useQuery(input, options);
```

The above is a little more verbose, but it allows a 1:1 swap of apis:

```tsx
const q = useCountQuery({ count: 1 });
const q2 = useCountQuery({ count: 1}, { transport, transformParams });
```

We could merge queryOptions and options but that creates potential for confusion if we ever introduce an option called `transformParams` to `useQuery` options so this is safest.

Reply:
Though honestly, merging the options objects should be fine. I think the chance for overlap is very slim.

Reply:
ah yeah, using 
```
inputs: [
            {
               ..variables,
            },
            {},
        ],
```
Kinda annoy and a bit ugly. Let me do adjust

Reply:
@paul-sachs for `queryOptions` is it okay to pass exactly type from `react-query` instead our custom type `{ transformParams }`

EX i'll look like
```
useCountQuery({count: 1}, {transport}, {onError, onSuccess})
```

Reply:
yeah. I got your idea. I predicted `queryOptions` be use more than `options`. So place it to second parameter is better, I think

Reply:
@paul-sachs I updated. could u help review